src/components/search/ResultItem.js

```diff
@@ -0,0 +1,341 @@
+    const handleMouseDown = () => {
+        nodeRef.current.willFocus = !(nodeRef.current.tUp && timeMs() - nodeRef.current.tUp < 300)
+    }
+    const handleMouseMove = () => {
+        nodeRef.current.willFocus = false
+    }
+    const handleMouseUp = () => {
+        if (nodeRef.current.willFocus) {
+            nodeRef.current.tUp = timeMs()
+            onPreview(url)
+        }
+    }
+
+    useEffect(() => {
+        if (isPreview && 'scrollIntoView' in nodeRef.current) {
+            nodeRef.current.scrollIntoView({ behavior: 'smooth', block: 'center' })
+        }
+    }, [isPreview])
+
+    const fields = hit._source || {}
+    const highlights = hit.highlight || {}
+    const downloadUrl = createDownloadUrl(url, fields.filename)
+
+    const cardHeaderClasses = {
+        action: classes.cardHeaderAction,
+        content: classes.cardHeaderContent,
+    }
+    const cardContentClasses = {
+        root: classes.cardContentRoot
+    }
+
+    return (
+        <Card
+            ref={nodeRef}
+            className={cn(classes.card, { [classes.selected]: isPreview })}
+            onMouseDown={handleMouseDown}
+            onMouseMove={handleMouseMove}
+            onMouseUp={handleMouseUp}
+        >
+            <CardHeader
+                classes={cardHeaderClasses}
+                title={
+                    <span className={classes.title}>
+                        {index}. {fields.filename}
+                    </span>
+                }
+                subheader={
+                    <span className={classes.title}>
+                        {truncatePath(fields.path)}
+                    </span>
+                }
+                action={
+                    <>
+                        <Grid container alignItems="center">
+                            <Grid item>
+                                <Tooltip title="Download original file">
+                                    <IconButton size="small">
+                                        <a href={downloadUrl} className={classes.buttonLink}>
+                                            <CloudDownloadOutlined className={classes.actionIcon} />
+                                        </a>
+                                    </IconButton>
+                                </Tooltip>
+                            </Grid>
+
+                            <Grid item>
+                                <Tooltip title="Open in new tab">
+                                    <IconButton size="small" style={{ marginRight: 15 }}>
+                                        <a href={url} target="_blank" className={classes.buttonLink}>
+                                            <Launch className={classes.actionIcon} />
+                                        </a>
+                                    </IconButton>
+                                </Tooltip>
+                            </Grid>
+
+                            {fields[`priv-tags.${whoAmI.username}`]?.includes('important') && (
```

Comment:
same as above: please get the icon information from some list with tags configuration, so we don't have to edit this code here every time we want to change tag name / color / tooltips / fieldname, or if we want to make these configurable...

Reply:
configuration added